lib/ansible/module_utils/compat/ipaddress.py

```diff
@@ -346,7 +346,10 @@ def _find_address_range(addresses):
 
     """
     it = iter(addresses)
-    first = last = next(it)
+    try:
+        first = last = next(it)
+    except StopIteration:
+        return
```

Comment:
Since this is in a vendored file, should we really be updating it like this?

We have history of applying upstream patches to our vendored libs, but not modifying them to diverge.

Reply:
Good catch. This is only a bug when calling `_find_address_range` directly, which is probably why it isn't fixed upstream. Only `collapse_addresses` calls it, and never with an empty list, which is needed to trigger the bug on Python 3.7 and later.

If we're OK with telling people not to call `_find_address_range` directly we could just add an ignore to `test/sanity/pylint/ignore.txt` with a comment stating that we don't intend to fix the issue. All the other ignores listed are intended to be fixed at some point.

@sivel @abadger What do you think?

Reply:
I agree we shouldn't make a change to the logic of the code.  I'm okay with whichever methods of ignoring the issue in this file that other people like, whether an ignore.txt or a pylint disable comment or etc